common.blocks/i-bem/__dom/i-bem__dom.js

```diff
@@ -736,13 +736,23 @@ DOM = BEM.decl('i-bem__dom',/** @lends BEMDOM.prototype */{
             modName = undef;
         }
 
+        names = names.split(' ');
+
         var _self = this.__self,
-            selector = '.' +
-                names.split(' ').map(function(name) {
-                    return _self.buildClass(name, modName, modVal);
-                }).join(',.'),
+            keys = names.map(function(name) {
+                return _self.buildClass(name, modName, modVal);
+            }),
+            isSingleName = keys.length === 1,
+            selector = '.' + (isSingleName? keys[0] : keys.join(',.')),
             res = findDomElem(ctx, selector);
 
+        // caching results if possible
+        (ctx === this.domElem) && keys.forEach(function(key, i) {
```

Comment:
As far as I remember, it's not save to compare jQuery-nodes in this way. Check this:

~~~js
var link === $('.link:eq(0)');
link === $(link)
~~~

Can you compare underlying DOM-nodes instead?